pylops_mpi/basicoperators/BlockDiag.py

```diff
@@ -0,0 +1,139 @@
+import numpy as np
+from scipy.sparse.linalg._interface import _get_dtype
+from mpi4py import MPI
+from typing import Optional, Sequence
+
+from pylops.utils import DTypeLike
+from pylops import LinearOperator
+
+from pylops_mpi import DistributedArray
+
+
+class MPIBlockDiag(LinearOperator):
+    r"""MPI Block-diagonal operator.
+
+        Create a block-diagonal operator from N linear operators using MPI.
+
+        Parameters
+        ----------
+        ops : :obj:`list`
+            Linear operators to be stacked. Alternatively,
+            :obj:`numpy.ndarray` or :obj:`scipy.sparse` matrices can be passed
+            in place of one or more operators.
+        base_comm : :obj:`mpi4py.MPI.Comm`, optional
+            Base MPI Communicator. Defaults to ``mpi4py.MPI.COMM_WORLD``.
+        dtype : :obj:`str`, optional
+            Type of elements in input array.
+
+        Attributes
+        ----------
+        shape : :obj:`tuple`
+            Operator shape
+        explicit : :obj:`bool`
+            Operator contains a matrix that can be solved explicitly (``True``) or not (``False``)
+
+        Notes
```

Comment:
Very good :D

Reply:
There is also a slight inconsistency with the fact that we allow each rank to have not just one operator but a list of operators... let's maybe simply add.

```
Each rank has one or more :class:`pylops.LinearOperator`, which we represent here compactly as :math:`\mathbf{L}_i` for rank :math:`i`.

Reply:
Finally

**L** -> :math:`\mathbf{L}`

Reply:
Done :slightly_smiling_face: